engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/result/json/JsonNodeExtractor.kt

```diff
@@ -28,6 +30,34 @@ object JsonNodeExtractor {
         }
     }
 
+    /**
+     * Extract the node at the given json node path.
+     */
+    fun getNodeAt(data: JsonMap, path: JsonNodePath): JsonNode? {
+        val rootNode = JsonNode(JsonNodePath.root, data)
+        return getNodeAt(rootNode, path)
+    }
+
+    /**
+     * Extract the node at the given json node path.
+     */
+    fun getNodeAt(rootNode: JsonNode, path: JsonNodePath): JsonNode? {
+        return path.segments.fold(rootNode as JsonNode?) { currentNode, segment ->
+            when (currentNode?.value) {
+                is Map<*, *> -> currentNode.value[segment.value]?.let {
+                    JsonNode(currentNode.resultPath + segment, it)
+                }
+                is List<*> -> when (segment.value) {
+                    is Int -> currentNode.value.getOrNull(segment.value as Int)?.let {
```

Comment:
Not sure if we should strictly cast this to `Int`. Perhaps we should just return null under the assumption that we can't find a JSON node instead of erroring out?

Reply:
I've changed the pattern matching expression to use `segment` instead of `value`, that way I don't need to cast anything 👍